Native list handling

Dysproz opened the pull request:
I added native list handling - there's no need to use list_embedded flag or dotty_l function.

Dotty now uses it's dictionary stored in self._data that already supports lists.
New architecture support even really deeply stored lists.

Although, I slightly refactored the whole code to be shorter and (hopefully) easier to read. :)

I added also warning that will appear when someone will try to use embedded list flag that it's depreciated. It can be removed after some time with flag itself and dotty_l function.

Also - It seems that python returns true for isdigit() function only when there's integer under string so there's no need to do the try/except thing with conversion.

pawelzny commented:
Hi @Dysproz 
Thank you for this pull request. I have only one request for you. Can you add new example with multidimensional lists ?
Dotty has already example with flat list and i think it would be cool to cover nested lists as well.

You can find flat list example here: https://github.com/pawelzny/dotty_dict/blob/master/example/advanced.py#L60

Dysproz commented:
@pawelzny
Example updated.
I added another list with some nested lists.
Although, I added info in comment that dotty_l is depreciated and dotty support embedded lists natively

pawelzny commented on example/advanced.py:
```diff
@@ -58,24 +58,38 @@ def make_request(payload):
-        ]
+        ],
+        'spec': {
+            'containers': [
```
IMO it will be more valuable to check if we can access lists in lists.
Mixed dict and list works right now without any changes. This PR provides a way to access lists in lists and this fact should be more emphasize.

The example can be changed to:

```python
'containers': [
    ['gpu', 'tensorflow', 'ML'],
    ['cpu', 'webserver', 'sql'],
]
```

And then assertions :

```python
assert dot['spec.containers.0.0'] == 'gpu'
assert dot['spec.containers.0.1'] == 'tensorflow'
assert dot['spec.containers.0.2'] == 'ML'
assert dot['spec.containers.1.0'] == 'cpu'
assert dot['spec.containers.1.1'] == 'webserver'
assert dot['spec.containers.1.2'] == 'sql'
```

Dysproz replied:
Resolved

Dysproz commented:
@pawelzny 
Is it how you requested it? :)
